### ISSUE TITLE
Requesting files from a holding that doesn't exist results in perpetual "ROUTING"

### ISSUE THREAD

**Issue by nmassey001** [cedadev/nlds#56]
Requires transaction level retries.
```
# get the holding from the database
        if holding_label is None and holding_id is None:
            holding = None
        else:
            try:
                holding = self.catalog.get_holding(
                    user, group, holding_label, holding_id
                )
            except CatalogError as e:
                self.log(e.message, RMQC.RK_LOG_ERROR)
                return
```
lines 324-334 of catalog_worker.py

**Comment by jackleland** [comment]
Can now be addressed with a transaction-level retry, following bb9de4cb631ee754a3affa22af1c95307e578f73

**Comment by jackleland** [comment]
Addressed as of 452dedc
